functions/Get-DbaMsdtc.ps1

```diff
@@ -71,17 +71,21 @@ Get-DbaMsdtc -ComputerName srv0042 | Out-Gridview
 
 Get DTC status for the computer srv0042 and show in a grid view
 
+
+
 #>
 	
 	[CmdletBinding()]
 	Param (
-		[Parameter(Mandatory = $true, ValueFromPipeline = $true)]
+		[Parameter(ValueFromPipeline = $true)]
 		[Alias("cn", "host", "Server")]
-		[string[]]$Computername
+		[string[]]$Computername = $env:COMPUTERNAME
 	)
 	
-	BEGIN
+BEGIN
 	{
+    $functionName = "Get-DbaMsDtc"
+    $ComputerName = $ComputerName | ForEach-Object {$_.split("\")[0]} | Select-Object -Unique

```

Comment:
Well, in that case you may want to change it up a little:
Simply put, it isn't going to work when used with a pipeline (Since it runs begin only once on the first object piped in).

Instead:
In Begin, create a list of computernames already processed
During Process, within the foreach loop, add the computer to that list if it isn't already tehre or skip it.
This way it works well in both modes